### DISCUSSION TITLE
Zoom in/out

### DISCUSSION THREAD

**Discussion by sicmundu** [google/model-viewer#1889] [Q&A]
Is there any way to control the zoom in/out through html elements? I tried to do through scrollTop on jquery, but failed.
Thanks in advance

**Comment by sicmundu** [comment]
Well, I made this by click on element.

```
jQuery("#plus").click(function(){
const modelViewer = document.querySelector('model-viewer');
modelViewer.cameraOrbit = 'auto auto 10%;
});
```
but zooming don't work as I want

**Reply by MrSounds** [reply]
Did you find a solution for zooming with an HTML-element?

**Reply by elalish** [reply]
We just added a feature in v1.10 that makes programmatic zooming much better and easier: https://modelviewer.dev/docs/#entrydocs-stagingandcameras-methods-zoom

**Reply by MrSounds** [reply]
Thank you for responding so fast!
I did also find a solution using FOV. This is my code in case anyone else might get some help from it:

let fovNumber = 90;

$(".bi-arrow-up").click(function() {
  if (fovNumber >=15){
    fovNumber = fovNumber - 5;
  };
  $("#mw").attr("field-of-view", fovNumber + "%");
});

$(".bi-arrow-down").click(function() {
  if (fovNumber <=90){
    fovNumber = fovNumber + 5;
  };
  $("#mw").attr("field-of-view", fovNumber + "%");
});

**Comment by elalish** [comment]
@grodstrike If you can be a little more specific about what you're trying to achieve, we may be able to better help. I think you're generally on the right track, except that we specify absolute numbers, not relative to your current position, so you'll have to do a little more math. Also, if the radius of `cameraOrbit` doesn't give you what you want, you can also zoom using `fieldOfView`.

**Comment by schmidtsonian** [comment]
Is there a way of preventing zooming?

**Comment by elalish** [comment]
@schmidtsonian Yes, if you set the `min-camera-orbit` and `max-camera-orbit` radius to the same value and also set `min-field-of-view` and `max-field-of-view` to the same value then no zooming will happen.

**Comment by schmidtsonian** [comment]
@elalish I made this: https://codepen.io/schmidtsonian/pen/VwvEwVw?page=1, and is still zooming

**Comment by elalish** [comment]
@schmidtsonian Ah, it's because your model-viewer is very out of date and doesn't even recognize these attributes. You're using 0.6 and our latest is 0.10. Try that and see if it works better. Also, `camera-orbit` is has three parameters, theta, phi and radius, so you'd want something more like `auto auto 1m`. For FoV, maybe start with `45deg`.

**Comment by nick-sp** [comment]
How can one use the latest model-viewer on a website?

Do you have to change something in these lines?

`<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.js"></script><script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
`
I changed them to
`<script type="module" src="https://unpkg.com/@google/model-viewer@0.10.0/dist/model-viewer.js"></script><script nomodule src="https://unpkg.com/@google/model-viewer@0.10.0/dist/model-viewer-legacy.js"></script>`

But I still can't set zoom and rotation limits using min/max camera orbit.

**Comment by elalish** [comment]
@nick-sp that should work fine. You can test by navigating to that URL and see that it automatically forwards to the latest version: `model-viewer@0.10.0`. The problem with the example above was that they explicitly chose `model-viewer@0.6.0`.

**Comment by elalish** [comment]
@nick-sp Can you post a Glitch or codepen I can look at?

**Comment by nick-sp** [comment]
Thank you for the response. I'm trying to lock the rotation to the vertical axis using this:

min-camera-orbit='auto 0m auto'
max-camera-orbit='auto 0m auto'

But it doesn't work

**Comment by elalish** [comment]
@nick-sp Ah, I think the problem is you're specifying an angle with meters. Try 'auto 75deg auto', since 75 is our default phi.

**Comment by Turtleted21** [comment]
Hello
modelViewer.cameraFieldOfView not exist?
so how can i set FieldOfView in a function?

Thanks

edit found it : modelViewer.fieldOfView

**Comment by ParadonPokpingmaung** [comment]
use zoom()
 
 const modelViewer = document.querySelector('.my-model');
  const zoomOutButton = document.querySelector('#zoom-out-button');
  const zoomInButton = document.querySelector('#zoom-in-button');

  zoomInButton.addEventListener('click', () => {
      modelViewer.zoom(2);
  });

  zoomOutButton.addEventListener('click', () => {
      modelViewer.zoom(-2);
  });
